doc/content/getting_started/contributing.md

```diff
@@ -0,0 +1,244 @@
+   cd ~/projects/tmap8
+   ```
+
+1. It is usually best to make sure your changes are based on devel:
+
+   ```
+   git checkout devel
+   ```
+
+1. Clean and reset devel to get it up to date (make sure you do not have any changes on devel, because you would lose that):
+
+   ```
+   git reset --hard upstream/devel && git submodule update --init
+   ```
+
+1. Create new branch:
+
+   ```
+   git checkout -b <BRANCH_NAME>
+   ```
+
+## Make modifications and commit (the regular development workflow)
+
+At this point, you have a local branch that is up to date with the upstream remote. You’re now ready
+to make your changes. Remember, **small, consistent changes** are much better than dumping massive,
+unrelated changes all at once.
+
+1. Make your changes locally to address the issue (partly or fully) - using
+   [VSCode is recommended](help/development/VSCode.md).
+1. In your terminal, go to your application:
+
+   ```
+   cd ~/projects/tmap8
+   ```
+
+1. Make sure that the app still compiles and that the tests run:
+
+   ```
+   make -j6 && ./run_tests -j6
+   ```
+
+   If you modified the documentation, you can then build and test it locally:
+
+   ```
+   cd ~/projects/tmap8/doc && ./moosedocs.py build --serve
+   ```
+
+1. Type `git status` to see the status of your branch. That should show you the changed files and
+   give you some commands to stage them.
+1. Once you are ready, perform
+
+   ```
+   git add <FILE_NAME_1> <FILE_NAME_2> <FILE_NAME_3> ...
+   ```
+
+   to stage your updated files.
+1. At that point, it is usually good practice to run
+
+   ```
+   git clang-format HEAD~1
+   ```
+
+   to fix the formatting of the files based on the `.clang-format` file in the root of the repository.
+   If files are changed, run
+
+   ```
+   git add <...>
+   ```
+
+   for these files.
+1. Now you’re ready to commit. Run
+
+   ```
+   git commit
+   ```
+
+1. A file editor will appear. Press the `i` key, and you’ll be able to type a short description of
+   the commit. It is good practice to have a short title, and then a list of items describing the
+   changes in more detail. At the bottom, you should add `(Ref. #<ISSUE_NUMBER>)`. It will look
+   something like this:
+
+   ```
+   Updating this to enable that:
+   - Update this file to do this
+   - Update documentation to reflect that change
+   - Create new test for new capability
+   - Update existing gold files to reflect change
+
+   (Ref. #1234)
+   ```
+
+1. To exit and save, press `esc`, then type `:wq`. The commit file will close.
+1. You might have to do several commits to capture all your changes.
+
+## Generating new documentation pages using MooseDocs
+
+During the course of development, especially during the creation of new TMAP8 objects (e.g., kernels,
+boundary conditions, interface conditions, etc.), it is important to create documentation outlining
+any new capabilities. Templates for object documents can be generated using the [MooseDocs system](MooseDocs/index.md)
```

Comment:
Add `${MOOSE_DIR}/python/doc/content` to the content section of `config.yml` to enable this.